### ISSUE TITLE
Raises ProgammingError, SA expects IntegrityError

### ISSUE THREAD

**Issue by srprogrammer** [joshuahlang/sqlalchemy-ctds#1]
Trying to fix some of the failures that happen when the SA [dialect compliance suite](https://github.com/zzzeek/sqlalchemy/blob/master/README.dialects.rst)  is run against sqlalchemy-cts.

Started with the test_dialect module.

One test which fails is [`tests/test_suite.py::ExceptionTest_mssql+ctds_14_0_3015_40::()::test_integrity_error`](https://github.com/zzzeek/sqlalchemy/blob/1210f49bb5a691324a55b4c3bf4600c359f396a9/lib/sqlalchemy/testing/suite/test_dialect.py#L29-L47)

SA expects `IntegryError` to get raised when a duplicate PK insert is attempted. See https://github.com/zzzeek/sqlalchemy/blob/e72cf013cd11cc77de5518c1fe66c9b8b42c3548/lib/sqlalchemy/testing/requirements.py#L448-L454

 Apparently most DB-API drivers do that.  Even sqlalchemy-ctds currently declares so by not overriding that `duplicate_key_raises_integrity_error` property in [requirements.py](https://github.com/joshualang/sqlalchemy-ctds/blob/master/src/sqlalchemy_ctds/requirements.py) to return `exclusions.closed()` (which would simply cause the test case to be skipped.)

ctds raises  `ProgrammingError`. It even has an test which involves such scenario: https://github.com/zillow/ctds/blob/54ba5842f0b7fce944883e519279c5723a643549/tests/test_cursor_execute.py#L191-L231

Actually don't know what would be the best path to follow. Hopefully this info will help you to make an informed decision.

**Comment by joshuahlang** [comment]
I would agree that should raise `ctds.IntegrityError`, though the DB API specification is a bit lacking in detail as to what falls into this category. It appears pymssql just [hardcodes some error numbers](https://github.com/pymssql/pymssql/blob/3bc299936fe2d35196dd4c9ebc8a22777f0c97e6/src/pymssql.pyx#L60-L64). I suppose that is sufficient to get SQLAlchemy functioning to the same level as it does with pymssql, though not the cleanest solution. Of course there doesn't seem to be a better one either...

**Comment by joshuahlang** [comment]
Depends on https://github.com/zillow/ctds/issues/12
